### PR TITLE
702.document snapshots

### LINKED ISSUE #702: Document the "snapshot" format
The format of the persistent on-grid state created by magic-folder might be written and read by different implementations (maybe different versions of magic-folder itself! or maybe entirely different implementations).

It would be helpful to have a document which serves as the canonical reference for this format, rather than having a certain revision of this implementation be the reference for this format.

This would help different implementations interoperate.  It might also help compatibility of magic-folder with itself over time.

### PR DESCRIPTION
**Pull request by meejah**
Fixes #702

### CODE REVIEW

**Comment by meejah** [comment]
Note to self: the `metadata` entries in the Collective and Personal should be `@metadata` instead.

**Comment by meejah** [comment]
I tried ensuring all the SVGs are "plain" (not "Inkscape") SVGs but github / firefox is still rendering the arrows w/o heads.

That's the only gnit though; I'm going to merge this -- please file followup tickets if anything is missing or unclear :)

File: docs/datamodel.rst [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+All read-write Participants hold the single Write Capability corresponding to their public Read-Only Directory Capability.
+We call these a "**Personal**" directory.
+
+Here is an example: we have a Folder called "My Notes" whose Collective contains two Participants named "laptop" and "desktop".
+The ``"@metadata"`` Immutable contains JSON describing the version (there is only one version right now, ``1``).
+
+.. figure:: magic-folder-data-model--high-level.svg
+    :width: 100%
+    :height: 30pt
```
[the remark this answers is not in the document]
Comment (hacklschorsch) [reply docs/datamodel.rst:71]: (On my end, a height of 300pt looks good in GitHub)

[the remark this answers is not in the document]
Comment (meejah) [reply docs/datamodel.rst:71]: https://github.com/LeastAuthority/magic-folder/pull/730

**Pull request merged**